### PR TITLE
Update suppression guidance on ASP0014

### PR DESCRIPTION
**Pull request by captainsafia**
Close https://github.com/dotnet/AspNetCore.Docs/issues/28364.

### CODE REVIEW

File: aspnetcore/diagnostics/asp0014.md [hunk 1]
```diff
@@ -51,4 +51,17 @@ app.Run();
 
 ## When to suppress warnings
 
-Do ***not*** suppress a warning from this rule.
+Warnings from this rule can be supressed if the target `UseEndpoints` invocation is invoked without any mappings as a strategy to organize middleware ordering.
```
Comment (Rick-Anderson) [inline aspnetcore/diagnostics/asp0014.md:54]:
```suggestion
Warnings from this rule can be suppressed if the target `UseEndpoints` doesn't impact middleware ordering, as shown in the following code:
```

**Review by Rick-Anderson — CHANGES_REQUESTED** [review]

**Comment by Rick-Anderson** [comment]
[Internal review URL](https://review.learn.microsoft.com/en-us/aspnet/core/diagnostics/asp0014?view=aspnetcore-8.0&branch=pr-en-us-28438)
Closes https://github.com/dotnet/AspNetCore.Docs/issues/28364.

**Comment by Rick-Anderson** [comment]
@iSOcH please review. Is my rewording correct?

File: aspnetcore/diagnostics/asp0014.md (hunk 1 above)
Comment (iSOcH) [reply aspnetcore/diagnostics/asp0014.md:54]: but `.UseEndpoints(_ => {})` is used to impact middleware ordering.

Comment (Rick-Anderson) [reply aspnetcore/diagnostics/asp0014.md:54]: > but `.UseEndpoints(_ => {})` is used to impact middleware ordering. see the referenced issue #28364

Thanks, reverted.

**Review by Rick-Anderson — APPROVED** [review]

**Comment by Rick-Anderson** [comment]
@captainsafia  can I S&M?

**Pull request merged**